modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt

```diff
@@ -286,6 +286,93 @@ class InstructionsTest {
         }
     }
 
+    @Test
+    @WithIroha
+    fun `burn and mint public key instruction committed`(): Unit = runBlocking {
+        // check Bob's public key before burn it
+        val bobPubKey = BOB_KEYPAIR.public.toIrohaPublicKey()
+        val query = QueryBuilder.findAccountById(BOB_ACCOUNT_ID)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+        val signatories = client.sendQuery(query).signatories
+        assertEquals(1, signatories.size)
+        assertContentEquals(bobPubKey.payload, signatories.first().payload)
+
+        // burn Bob's public key
+        client.sendTransaction {
+            account(ALICE_ACCOUNT_ID)
+            burnPublicKey(BOB_ACCOUNT_ID, bobPubKey)
+            buildSigned(ALICE_KEYPAIR)
+        }.also {
+            Assertions.assertDoesNotThrow {
+                it.get(10, TimeUnit.SECONDS)
+            }
+        }
+
+        // generate new key pair without salt for Bob's account
+        val newKeyPair = generateKeyPair()
+
+        // change Bob's account public key
+        client.sendTransaction {
+            account(ALICE_ACCOUNT_ID)
+            mintPublicKey(BOB_ACCOUNT_ID, newKeyPair.public.toIrohaPublicKey())
+            buildSigned(ALICE_KEYPAIR)
+        }.also {
+            Assertions.assertDoesNotThrow {
+                it.get(10, TimeUnit.SECONDS)
+            }
+        }
+
+        // check new public key in Bob's account
+        val newPubKeyQuery = QueryBuilder.findAccountById(BOB_ACCOUNT_ID)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+        val signatoriesWithNewPubKey = client.sendQuery(newPubKeyQuery).signatories
+        assertEquals(1, signatoriesWithNewPubKey.size)
+        assertContentEquals(newKeyPair.public.toIrohaPublicKey().payload, signatoriesWithNewPubKey.first().payload)
+    }
+
+    @Test
+    @WithIroha
+    fun `change user account metadata`(): Unit = runBlocking {
+        // grant permission to Alice to change Bob's account metadata
+        client.sendTransaction {
+            account(BOB_ACCOUNT_ID)
+            grantSetKeyValueAccount(BOB_ACCOUNT_ID, ALICE_ACCOUNT_ID)
+            buildSigned(BOB_KEYPAIR)
+        }.also {
+            Assertions.assertDoesNotThrow {
+                it.get(10, TimeUnit.SECONDS)
+            }
+        }
+
+        // check permission
+        val permissionQuery = QueryBuilder.findAccountById(ALICE_ACCOUNT_ID)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+        val permissionTokens = client.sendQuery(permissionQuery).permissionTokens
+        assertEquals(1, permissionTokens.size)
+
+        // add\update salt value in Bob's account metadata
+        val salt = "ABCDEFG".asValue()
+        client.sendTransaction {
+            account(ALICE_ACCOUNT_ID)
+            setKeyValue(BOB_ACCOUNT_ID, "salt".asName(), salt)
```

Comment:
Please lift `"salt"` to a const at the beginning of the test

Reply:
done